src/main/java/org/mule/extension/db/internal/domain/type/ResolvedDbType.java

```diff
@@ -29,21 +30,26 @@ public ResolvedDbType(int id, String name) {
   @Override
   public void setParameterValue(PreparedStatement statement, int index, Object value, DbConnection connection)
       throws SQLException {
-    if (value == null) {
-      statement.setNull(index, id);
-    } else {
+    try {
+      if (value == null) {
+        statement.setNull(index, id);
+        return;
+      }
+
       if (DECIMAL == id || NUMERIC == id) {
         if (value instanceof BigDecimal) {
           statement.setObject(index, value, id, ((BigDecimal) value).scale());
+          return;
         } else if (value instanceof Float || value instanceof Double) {
           BigDecimal bigDecimal = new BigDecimal(value.toString());
           statement.setObject(index, bigDecimal, id, bigDecimal.scale());
-        } else {
-          statement.setObject(index, value, id);
+          return;
         }
-      } else {
-        statement.setObject(index, value, id);
       }
+
+      statement.setObject(index, value, id);
+    } catch (Exception e) {
+      statement.setObject(index, value, OTHER);
```

Comment:
What does it happen when an exception is thrown here? I mean without this change
Can we break anything adding the parameter even though an exception has been thrown?
Is it possible that the we get a different exception because of this change?

Reply:
Right now it's throwing a mule exception of UNKNOWN type. All this does is try to delegate the task of casting this parameter to the engine and then a DB:QUERY_EXECUTION with description Invalid column type is thrown, if the parameter cannot be cast.